src/lib.rs

```diff
@@ -4,41 +4,133 @@
+//! to enable you to opt-in to additional functionality. For a detailed description, see
+//! the [Features Reference chapter of the guide](https://pyo3.rs/main/features.html#features-reference).
+//!
+//! ## Default feature flags
+//!
+//! The following features are turned on by default:
+//! - `macros`: Enables various macros, including all the attribute macros.
+//!
+//! ## Optional feature flags
+//!
+//! The following features are optional:
+//! - `abi3`: Restricts PyO3's API to a subset of the full Python API which is guaranteed
+//! by [PEP 384](https://www.python.org/dev/peps/pep-0384/) to be forward-compatible with future Python versions.
+//
+//! - `auto-initialize`: Changes [`Python::with_gil`](crate::Python::with_gil) and
+//! [`Python::acquire_gil`](crate::Python::acquire_gil) to automatically initialize the
+//! Python interpreter if needed.
+//
+//! - `extension-module`: This will tell the linker to keep the Python symbols unresolved,
+//! so that your module can also be used with statically linked Python interpreters.
+//! Use this feature when building an extension module.
+//
+//! - `hashbrown`: Enables conversions between Python objects and
+//! [hashbrown](https://docs.rs/hashbrown)'s
+//! [`HashMap`](https://docs.rs/hashbrown/latest/hashbrown/struct.HashMap.html) and
+//! [`HashSet`](https://docs.rs/hashbrown/latest/hashbrown/struct.HashSet.html) types.
+//
+//! - `multiple-pymethods`: Enables the use of multiple
+//! [`#[pymethods]`](crate::proc_macro::pymethods) blocks per
+//! [`#[pyclass]`](crate::proc_macro::pyclass). This adds a dependency on the
+//! [`inventory`](https://docs.rs/inventory) crate, which is not supported on all platforms.
+//
+//! - `num-bigint`: Enables conversions between Python objects and
+//! [num-bigint](https://docs.rs/num-bigint)'s
+//! [`BigInt`](https://docs.rs/num-bigint/latest/num_bigint/struct.BigInt.html) and
+//! [`BigUint`](https://docs.rs/num-bigint/latest/num_bigint/struct.BigUint.html) types.
+//
+//! - `num-complex`: Enables conversions between Python objects and
+//! [num-complex](https://docs.rs/num-complex)'s
+//! [`Complex`](https://docs.rs/num-complex/latest/num_complex/struct.Complex.html) type.
+//
+//! - `serde`: Allows implementing [serde](https://docs.rs/serde)'s
+//! [`Serialize`](https://docs.rs/serde/latest/serde/trait.Serialize.html) and
+//! [`Deserialize`](https://docs.rs/serde/latest/serde/trait.Deserialize.html) traits for
+//! [`Py`]`<T>` for all `T` that implement
+//! [`Serialize`](https://docs.rs/serde/latest/serde/trait.Serialize.html) and
+//! [`Deserialize`](https://docs.rs/serde/latest/serde/trait.Deserialize.html).
+//!
+//! ## Unstable features
+//!
+//! - `nightly`: Gates some optimizations that rely on
+//!  [`#![feature(specialization)]`](https://github.com/rust-lang/rfcs/blob/master/text/1210-impl-specialization.md),
+//! for which you'd also need nightly Rust. You should not use this feature.
```

Comment:
Fully agree 😂. I keep wondering if we should remove this feature!

Reply:
We might be able to substitute with `min_specialization`? It is less unsound than full `specialization`, at least...

I figured that since people can see the feature exists it should at least be mentioned with a note that you shouldn't use it.